ui/helpers/ramps/rampApi/rampAPI.ts

```diff
@@ -0,0 +1,24 @@
+import getFetchWithTimeout from '../../../../shared/modules/fetch-with-timeout';
+import { AggregatorNetwork } from '../../../ducks/ramps/types';
+import { defaultBuyableChains } from '../../../ducks/ramps/constants';
+
+const fetchWithTimeout = getFetchWithTimeout();
+
+const rampApiBaseUrl =
+  process.env.METAMASK_RAMP_API_BASE_URL ||
+  'https://on-ramp-content.metaswap.codefi.network';
+
+const RampAPI = {
+  async getNetworks(): Promise<AggregatorNetwork[]> {
+    try {
+      const url = `${rampApiBaseUrl}/regions/networks?context=extension`;
+      const response = await fetchWithTimeout(url);
+      const { networks } = await response.json();
+      return networks;
+    } catch (error) {
+      return defaultBuyableChains;
+    }
```

Comment:
Here we are mixing some concerns, the API object must be responsible of performing the API call only, by defaulting here we are overriding the expected behavior of the async thunk, which would never be called.

This method must not be wrapped in a try catch unless we want to log the error in the catch. It must throw just like the SDK or any other API call does.

Reply:
https://github.com/MetaMask/metamask-extension/commit/988e98509c129d77fb4127cbed2f94b974ee4718